Resources/public/js/views/ez-languageswitcherview.js

```diff
@@ -0,0 +1,158 @@
+/*
+ * Copyright (C) eZ Systems AS. All rights reserved.
+ * For full copyright and license information view LICENSE file distributed with this source code.
+ */
+YUI.add('ez-languageswitcherview', function (Y) {
+    "use strict";
+    /**
+     * Provides the Language Switcher View class
+     *
+     * @module ez-languageswitcherview
+     */
+    Y.namespace('eZ');
+
+    /**
+     * The raw content view
+     *
+     * @namespace eZ
+     * @class LanguageSwitcherView
+     * @constructor
+     * @extends eZ.TemplateBasedView
+     */
+    Y.eZ.LanguageSwitcherView = Y.Base.create('languageSwitcherView', Y.eZ.TemplateBasedView, [Y.eZ.Expandable], {
+        events: {
+            '.ez-dropdown-list-indicator': {
+                'tap': '_toggleExpanded'
+            }
+        },
+
+        initializer: function () {
+            this.after('expandedChange', this._setClickOutsideEventHandler);
+        },
+
+        /**
+         * Renders the language switcher view
+         *
+         * @method render
+         * @return {eZ.LanguageSwitcherView} the view itself
+         */
+        render: function () {
+            var container = this.get('container');
+
+            container.setHTML(this.template({
+                location: this.get('location').toJSON(),
+                currentTranslation: this.get('languageCode'),
+                otherTranslations: this._getOtherTranslationsList()
+            }));
+
+            return this;
+        },
+
+        /**
+         * Returns array with translations of content other than already being viewed.
+         * If conten doesn't have got other translations than the one already being viewed,
+         * then empty array is returned.
+         *
+         * @method _getOtherTranslationsList
+         * @private
+         * @return {Array} other translations
+         */
+        _getOtherTranslationsList: function () {
+            var translations = this._getTranslationsList(),
+                languageCode = this.get('languageCode'),
+                idx;
+
+            idx = Y.Array.indexOf(translations, languageCode);
```

Comment:
this could be better written with something like:
```js
var languageCode = this.get('languageCode');
 
return Y.Array.reject(this._getTranslationList(), function (value) {
    return value === languageCode;
});
```